pkg/daemon/daemon.go

```diff
@@ -1470,9 +1470,17 @@ func (dn *Daemon) removeUpdateInProgressTaint(ctx context.Context) error {
 	})
 }
 
-func (dn *Daemon) isLayeredNode() bool {
+func (dn *Daemon) shouldUpdateLayered() *CoreOSDaemon {
 	_, ok := dn.node.Annotations[constants.DesiredImageConfigAnnotationKey]
-	return ok
+	if ok {
+		// sanity check that we're on CoreOS. The controller checks if a node runs CoreOS before setting desired image
+		if dn.os.IsCoreOSVariant() {
+			return &CoreOSDaemon{dn}
+		} else {
```

Comment:
```suggestion
	
```
don't need the else you can just log and return nil

Reply:
Fixed